### ISSUE TITLE
Tutorials and Sample page Filter by not properly work for mobile screen.

### ISSUE THREAD

**Issue by zalabhavy** [flutter/website#11980]
### Describe the problem

The "Filter by" sidebar on the Tutorials and Sample pages is not functioning properly on mobile screens. Currently, there is no visible close icon, making it difficult for users to close the filter panel after opening it.

<img width="703" alt="Image" src="https://github.com/user-attachments/assets/1d4c32dd-dc7f-45b3-878c-87e0c9796292" />

### Expected fix

A close icon/button should be added to the "Filter by" wrapper on mobile screens to allow users to easily close the sidebar.

https://github.com/user-attachments/assets/235cf47c-1faf-42b9-b983-50a998d78a7b

### On which browser(s) did you experience this issue?

Chrome

### I would like to fix this problem.

- [x] I will try and fix this problem on docs.flutter.dev.

**Comment by zalabhavy** [comment]
@parlough @Sfshaza We need this or not ?

**Comment by parlough** [comment]
Sounds like a great improvement @zalabhavy! Thanks for the thoughtful issue.

I'd be happy to review a PR adding this.
